### PR TITLE
feat: Normalize VCS provider names to match backend

### PR DESCRIPTION
**Pull request by runningcode**
## Summary

Backend does not accept github.com, it only accepts github
Normalize VCS provider names.

Closes EME-303

## Examples

| Hostname | Before | After |
|----------|--------|-------|
| `github.com` | `github.com` | `github` |
| `gitlab.com` | `gitlab.com` | `gitlab` |
| `dev.azure.com` | `dev.azure.com` | `azure` |
| `git.mycompany.com` | `git.mycompany.com` | `mycompany` |

🤖 Generated with [Claude Code](https://claude.ai/code)

### CODE REVIEW

**Comment by runningcode** [comment]
I had assumed that this was all added by us, but it seems some of it is used elsewhere which could break some things. I’ll take a look.

**Comment by szokeasaurusrex** [comment]
> I had assumed that this was all added by us, but it seems some of it is used elsewhere which could break some things. I’ll take a look.

@runningcode What is the result of this investigation? Is this PR ready for review?

**Comment by runningcode** [comment]
It was affecting the release feature. I've simplified the PR to only affect the function that we use. Should be much easier to review now!

File: src/utils/vcs.rs [hunk 1]
```diff
@@ -201,12 +201,22 @@ impl VcsUrl {
...
 
+fn extract_provider_name(host: &str) -> String {
+    // Take just the part immediately before the last dot
+    let parts: Vec<&str> = host.split('.').collect();
+    if parts.len() >= 2 {
+        parts[parts.len() - 2].to_owned()
+    } else {
+        host.to_owned()
+    }
+}
```
[the remark this answers is not in the document]
Comment (szokeasaurusrex) [reply src/utils/vcs.rs:213]: @runningcode this seems like a valid concern, even if only an edge case, as URLs like `https://github.com./repo/url`, while unconventional, are still valid, as you can see from this link: https://github.com./getsentry/sentry-cli/.

My preference here would be to use a regex. Something like this could work, the first capturing group contains the part we would need to extract: `^([^\.]+)\.[^\.]+.?$`.

**Review by szokeasaurusrex — COMMENTED** [review]
I think Cursor's point is valid

File: src/utils/vcs.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (szokeasaurusrex) [reply src/utils/vcs.rs:213]: Let's also add a test for this case

File: src/utils/vcs.rs [hunk 2]
```diff
@@ -922,6 +929,64 @@ mod tests {
...
+            get_provider_from_remote("https://github.mycompany.com/user/repo"),
+            "mycompany"
+        );
+        assert_eq!(
+            get_provider_from_remote("https://source.developers.google.com/p/project/r/repo"),
+            "google"
+        );
+        // Test edge case with trailing dot in hostname
+        assert_eq!(
+            get_provider_from_remote("https://github.com./user/repo"),
```
Comment (runningcode) [inline src/utils/vcs.rs:982]: test case added

File: src/utils/vcs.rs [hunk 3]
```diff
@@ -207,6 +207,13 @@ impl VcsUrl {
     }
 }
 
+fn extract_provider_name(host: &str) -> String {
+    let trimmed = host.trim_end_matches('.');
```
Comment (runningcode) [inline src/utils/vcs.rs:211]: I had claude try to come up with the best solution that passes all the tests.
This code seems really simple but I’m not sure how it manages to pass all the tests. 😅

File: src/utils/vcs.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (runningcode) [reply src/utils/vcs.rs:213]: added and simplified the code

File: src/utils/vcs.rs [hunk 4]
```diff
@@ -207,6 +207,13 @@ impl VcsUrl {
...
+    let mut iter = trimmed.rsplit('.');
+    iter.next(); // skip TLD
+    iter.next().unwrap_or(trimmed).to_owned()
```
Comment (szokeasaurusrex) [inline src/utils/vcs.rs]: [nit] I believe you can do this for even more simplicity:

```suggestion
    trimmed.rsplit('.').nth(1).unwrap_or(trimmed)
```

See https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.nth

File: src/utils/vcs.rs [hunk 5]
```diff
@@ -207,6 +207,13 @@ impl VcsUrl {
     }
 }
 
+fn extract_provider_name(host: &str) -> String {
```
Comment (szokeasaurusrex) [inline src/utils/vcs.rs]:
```suggestion
fn extract_provider_name(host: &str) -> &str {
```

[optional] I'd suggest returning a `&str` here, and performing the copy to `String` in the calling code if it is needed. There is no reason why this code needs to copy the extracted slice, even if now we need a copy, in the future we may want to use this function, where we do not need a copy.

File: src/utils/vcs.rs [hunk 6]
```diff
@@ -922,6 +929,64 @@ mod tests {
...
+    #[test]
+    fn test_extract_provider_name() {
+        // Test basic provider name extraction
+        assert_eq!(extract_provider_name("github.com"), "github");
+        assert_eq!(extract_provider_name("gitlab.com"), "gitlab");
+        assert_eq!(extract_provider_name("bitbucket.org"), "bitbucket");
+
+        // Test edge case with trailing dots
+        assert_eq!(extract_provider_name("github.com."), "github");
+        assert_eq!(extract_provider_name("gitlab.com.."), "gitlab");
```
Comment (szokeasaurusrex) [inline src/utils/vcs.rs]: Having two consecutive dots in a URL is invalid, so we don't need to test this case imo (behavior can be left undefined)

```suggestion
```

**Review by szokeasaurusrex — APPROVED** [review]
Looks good, please address minor comments prior to merge

File: src/utils/vcs.rs (hunk 4 above)
Comment (runningcode) [reply src/utils/vcs.rs]: That is a neat party trick!
